pkg/steps/bundle_source.go

```diff
@@ -69,35 +70,56 @@ func (s *bundleSourceStep) run(ctx context.Context) error {
 	return handleBuild(ctx, s.buildClient, build, s.artifactDir)
 }
 
-func replaceCommand(pullSpec, with string) string {
-	return fmt.Sprintf("find . -type f -name \\*.yaml -exec sed -i 's?%s?%s?g' {} +", pullSpec, with)
+func replaceCommand(pullSpec, with string) []string {
+	sedSub := fmt.Sprintf("s?%s?%s?g", pullSpec, with)
+	return []string{`find`, `.`, `-type`, `f`, `-regex`, `.*\.\(yaml\|yml\)`, `-exec`, `sed`, `-i`, sedSub, `{}`, `+`}
 }
 
 func (s *bundleSourceStep) bundleSourceDockerfile() (string, error) {
 	var dockerCommands []string
-	dockerCommands = append(dockerCommands, "")
 	dockerCommands = append(dockerCommands, fmt.Sprintf("FROM %s:%s", api.PipelineImageStream, api.PipelineImageStreamTagReferenceSource))
-	is, err := s.imageClient.ImageStreams(s.jobSpec.Namespace()).Get(context.TODO(), api.StableImageStream, meta.GetOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to get stable imagestream: %w", err)
-	}
+	imagestreams := make(map[string]*imageapi.ImageStream)
 	for _, sub := range s.config.Substitutions {
-		replaceSpec, exists := util.ResolvePullSpec(is, sub.With, false)
+		imageSubSplit := strings.Split(sub.With, ":")
```

Comment:
We should use `DependencyParts` in order to provide users the same UX here as we do everywhere else they're asked to provide an image name.